ddtrace/internal/core/crashtracking.py

```diff
@@ -1,68 +1,159 @@
-    crashtracker.set_use_alt_stack(bool(crashtracker_config.use_alt_stack))
-    if crashtracker_config.stacktrace_resolver == "fast":
-        crashtracker.set_resolve_frames_fast()
-    elif crashtracker_config.stacktrace_resolver == "full":
-        crashtracker.set_resolve_frames_full()
+    # Another thing to note is that these tag keys and values can be bytes
+    # objects, so we need to ensure that they are converted to strings as
+    # PyO3 can't convert bytes objects to Rust String type.
+    tags = {ensure_text(k): ensure_text(v) for k, v in tags.items()}
+
+    return tags
+
+
+def _get_args(additional_tags: Optional[Dict[str, str]]):
+    # First check whether crashtracker_exe command is available
+    crashtracker_exe_py = os.path.join(os.path.dirname(__file__), "crashtracker_exe.py")
+    if not os.path.exists(crashtracker_exe_py) or not os.access(crashtracker_exe_py, os.R_OK):
+        print("Failed to find crashtracker_exe in the scripts directory")
+        return (None, None, None)
+
+    if crashtracker_config.stacktrace_resolver is None:
+        stacktrace_resolver = StacktraceCollection.Disabled
+    elif crashtracker_config.stacktrace_resolver == "fast":
+        stacktrace_resolver = StacktraceCollection.WithoutSymbols
     elif crashtracker_config.stacktrace_resolver == "safe":
-        crashtracker.set_resolve_frames_safe()
+        stacktrace_resolver = StacktraceCollection.EnabledWithSymbolsInReceiver
+    elif crashtracker_config.stacktrace_resolver == "full":
+        stacktrace_resolver = StacktraceCollection.EnabledWithInprocessSymbols
     else:
-        crashtracker.set_resolve_frames_disable()
+        # This should never happen, as the value is validated in the crashtracker_config
+        # module.
+        print(f"Invalid stacktrace_resolver value: {crashtracker_config.stacktrace_resolver}")
+        stacktrace_resolver = StacktraceCollection.EnabledWithInprocessSymbols
 
-    if crashtracker_config.stdout_filename:
-        crashtracker.set_stdout_filename(crashtracker_config.stdout_filename)
-    if crashtracker_config.stderr_filename:
-        crashtracker.set_stderr_filename(crashtracker_config.stderr_filename)
+    # Create crashtracker configuration
+    config = CrashtrackerConfiguration(
+        [],  # additional_files
+        crashtracker_config.create_alt_stack,
+        crashtracker_config.use_alt_stack,
+        5000,  # timeout_ms
+        stacktrace_resolver,
+        crashtracker_config.debug_url or agent_config.trace_agent_url,
+        None,  # unix_socket_path
+    )
 
-    # Add user tags
-    for key, value in crashtracker_config.tags.items():
-        add_tag(key, value)
+    # Create crashtracker receiver configuration
+    receiver_config = CrashtrackerReceiverConfig(
+        [],  # args
+        # Need to set PATH to find the right python binary
+        os.environ.copy(),  # env
+        crashtracker_exe_py, # path_to_receiver_binary
```

Comment:
@brettlangdon
Ran chmod +x on the crashtracker_exe.py file and removed it from installed command. 
Then, now we have to make sure that the script can find the right Python binary that has dd-trace-py installed in. When a venv is activated, it will prepend `PATH` with directories to look for `python` binary, so I believe we need to pass `PATH` envvar. For our tests, that was sufficient, but I don't know what else needs to be passed. So simply copied over `os.environ`

Reply:
Do we want to explicitly use `sys.executable`?

Reply:
That didn't work. 

`sys.executable` returned `/root/project/.riot/venv_py3131_deps/bin/python` which didn't have `ddtrace` installed
```
root@docker-desktop:~/project#  /root/project/.riot/venv_py3131_deps/bin/pip list
Package Version
------- -------
pip     25.0.1
```

What we need is `/root/project/.riot/venv_py3131/bin/python` with

```
root@docker-desktop:~/project# /root/project/.riot/venv_py3131/bin/pip list | grep ddtrace
ddtrace            3.8.0.dev86+g1a733a1f1.d20250508 /root/project
```

Reply:
oh... weird

Reply:
weird, were you passing the right `PYTHONPATH` to the subprocess? since `riot` just hacks the `PYTHONPATH` to toss the "base" venv into it so `ddtrace` can be found.

Reply:
Would be similar for SSI as well

Reply:
I didn't try passing `PYTHONPATH`. I've only tried passing `PATH`, and it worked, but I feared whether there would be other things affecting how the script find the right Python and `ddtrace`, I just ended up passing all env vars.

Reply:
I am surprised `os.environ.copy()` doesn't "just work" 🤔

Reply:
Oh, that's what I'm using now and it works.